v2/pkg/runner/runner.go

```diff
@@ -534,8 +534,15 @@ func (r *Runner) RawSocketEnumeration(ip string, p *port.Port) {
 		gologger.Debug().Msgf("Skipping cdn target: %s:%d\n", ip, p.Port)
 		return
 	}
-
+	if r.scanner.ScanResults.IPHasPort(ip, p) {
+		return
+	}
 	r.limiter.Take()
+	open, err := r.scanner.ConnectPort(ip, p, time.Duration(r.options.Timeout)*time.Millisecond)
```

Comment:
This routine perform asyncronous SYN scan to avoid the complete TCP handshake, no full connection should be performed